File: lib/java/opentoken/src/main/java/com/truveta/opentoken/tokens/definitions/T6Token.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokens.definitions;

import java.util.ArrayList;

import com.truveta.opentoken.attributes.AttributeExpression;
import com.truveta.opentoken.tokens.Token;

/**
 * Represents optional T6 token definition.
 */
public class T6Token implements Token {
    private static final long serialVersionUID = 1L;
    private static final String ID = "T6";

    private final ArrayList<AttributeExpression> definition = new ArrayList<>();

    @Override
    public String getIdentifier() {
        return ID;
    }

    @Override
    public ArrayList<AttributeExpression> getDefinition() {
        return definition;
    }
}

File: lib/python/opentoken-core-ai/src/main/opentoken_core_ai/tokens/t6_token.py
"""Optional T6 token definition."""
from typing import List

from opentoken.attributes.attribute_expression import AttributeExpression
from opentoken.tokens.token import Token


class T6Token(Token):
    """Represents optional T6 token definition."""

    ID = "T6"

    def __init__(self):
        self._definition: List[AttributeExpression] = []

    def get_identifier(self) -> str:
        return self.ID

    def get_definition(self) -> List[AttributeExpression]:
        return self._definition
